zeroconf/_dns.py

```diff
@@ -257,12 +266,15 @@ def write(self, out: 'DNSOutgoing') -> None:
     def __eq__(self, other: Any) -> bool:
         """Tests equality on address"""
         return (
-            isinstance(other, DNSAddress) and DNSEntry.__eq__(self, other) and self.address == other.address
+            isinstance(other, DNSAddress)
+            and DNSEntry.__eq__(self, other)
```

Comment:
It would be faster to reject mismatched addresses sooner then checking eq.  I Will adjust in a follow up

Reply:
addressed here https://github.com/jstasiak/python-zeroconf/pull/808